src/hooks/usePromise.ts

```diff
@@ -16,49 +16,59 @@ export type PromiseType<T = any> = {
   error: any;
 };
 
+export type PromiseOptions = {
+  /**
+   * Determines whether to execute the promise automatically. Defaults to `true`. When set to `false`, executing the
+   * promise will be delayed until `enabled` is set to `true`.
+   */
+  enabled?: boolean;
+};
+
 /**
  * Hook that provides state management for asynchronous operations, representing the loading state,
  * the resolved data, and any error that may occur. It is particularly useful for handling promises,
  * such as data fetching operations.
  *
  * @template T - The expected type of the data to be resolved by the promise.
  * @param promiseFn - A function that returns a promise, which resolves to the data of type `T`.
+ * @param options - Options to control the behavior of the promise.
  * @param initialValue - The initial state for the data before the promise resolves.
  * @param deps - An array of dependencies that, when changed, will trigger the promise function to be called again.
  * @returns An object containing the current state of the asynchronous operation: the loading status, the resolved data, and any error.
  */
 export function usePromise<T>(
   promiseFn: () => Promise<T>,
+  options: PromiseOptions,
```

Comment:
we should make this optional and use default options (here and everywhere else)
Because it is optional, it needs to move to the end

Reply:
Done, although kept it where it is, since `deps` is also optional.